migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/TaxonomyLookup.java

```diff
@@ -0,0 +1,64 @@
+package uk.nhs.digital.ps.migrator.model.hippo;
+
+import java.io.*;
+import java.util.*;
+import org.yaml.snakeyaml.Yaml;
+
+public class TaxonomyLookup {
+
+    private final ArrayList<String> taxonomyKeys = new ArrayList<>();
+    private final static String HIPPO_TAXONOMY_KEY = "hippotaxonomy:key";
+
+    public TaxonomyLookup()
+    {
+        setup();
+    }
+
+    private void setup() {
+
+        // Load taxonomy yaml from file
+        // Currently uses development taxonomy file, but upon migration will use
+        // the full taxonomy structure from Initial Import of Taxonomy
+        Yaml yaml = new Yaml();
+        File taxonomyYamlFile = new File("./repository-data/development/src/main/resources/hcm-content/content/taxonomies/publication_taxonomy.yaml");
+
+        HashMap<String, String> map = null;
+
+        try {
+            map = yaml.load(new FileInputStream(taxonomyYamlFile));
+
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        // Extract all hippotaxonomy:key from yaml file
+        ExtractKeys(map);
+    }
+
+    private HashMap ExtractKeys(HashMap map) {
+
+        // Apply to all the sub-maps as well
+        Iterator it = map.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry pair = (Map.Entry)it.next();
+
+            if (pair.getValue() instanceof HashMap) {
+               map.put(pair.getKey(), ExtractKeys((HashMap)pair.getValue()));
```

Comment:
Don't think you need to put the result of ExtractKeys into the map as it's not getting modified. Should just be able to do
`ExtractKeys((HashMap)pair.getValue());`

You can then make this method void and not return the map as we don't need to look at the return value